persistence/sql/batch/create.go

```diff
@@ -213,21 +264,55 @@ func Create[T any](ctx context.Context, p *TracerConnection, models []*T) (err e
 	}
 	defer rows.Close()
 
+	// MySQL, which does not support RETURNING, also does not have ON CONFLICT DO
+	// NOTHING, meaning that MySQL will always fail the whole transaction on a single
+	// record conflict.
+	if conn.Dialect.Name() == dbal.DriverMySQL {
+		return sqlcon.HandleError(rows.Close())
+	}
+
+	if options.partialInserts {
+		return handlePartialInserts(queryArgs, values, models, rows)
+	} else {
+		return handleFullInserts(models, rows)
+	}
+
+}
+
+func handleFullInserts[T any](models []*T, rows *sql.Rows) error {
 	// Hydrate the models from the RETURNING clause.
-	//
-	// Databases not supporting RETURNING will just return 0 rows.
-	count := 0
-	for rows.Next() {
+	for i := 0; rows.Next(); i++ {
 		if err := rows.Err(); err != nil {
```

Comment:
This check is incorrect AFAIK. `rows.Err()` should be checked after `rows.Next() == false`.

Reply:
fixed :)